src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -198,17 +196,29 @@ private async Task<dynamic> GetSiteInternal(string stampName, string siteName, s
 
         public override async Task<string> GetStampName(string subscriptionId, string resourceGroupName, string siteName)
         {
+            if (string.IsNullOrWhiteSpace(subscriptionId))
+                throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrWhiteSpace(resourceGroupName))
+                throw new ArgumentNullException(nameof(resourceGroupName));
+
             var siteObjects = await GetAdminSitesAsync(siteName);
-            var siteObject = siteObjects?
-                .Select(i => (JObject)i)?
-                .FirstOrDefault(j =>
-                    j.ContainsKey("Subscription") &&
-                    j["Subscription"].ToString().Equals(subscriptionId, StringComparison.InvariantCultureIgnoreCase) &&
-                    j.ContainsKey("ResourceGroupName") &&
-                    j["ResourceGroupName"].ToString().Equals(resourceGroupName, StringComparison.InvariantCultureIgnoreCase) &&
-                    (j.ContainsKey("StampName") || j.ContainsKey("InternalStampName")));
-
-            return siteObject?["InternalStampName"]?.ToString() ?? siteObject?["StampName"]?.ToString() ?? string.Empty;
+            if (siteObjects == null || !siteObjects.Any())
+                throw new Exception($"Could not get admin sites for site {siteName}");
+
+            var icic = StringComparison.InvariantCultureIgnoreCase;
+            var objects = siteObjects.Select(x => (JObject)x)
+                .Where(x => x.ContainsKey("Subscription") && x["Subscription"].ToString().Equals(subscriptionId, icic))
+                .Where(x => x.ContainsKey("ResourceGroupName") && x["ResourceGroupName"].ToString().Equals(resourceGroupName, icic));
+
+            var internalStampObjects = objects.Where(x => x.ContainsKey("InternalStampName") && !string.IsNullOrWhiteSpace(x["InternalStampName"].ToString()));
+            if (internalStampObjects.Any())
+                return internalStampObjects.First()["InternalStampName"].ToString();
+
+            var stampNameObjects = objects.Where(x => x.ContainsKey("StampName") && !string.IsNullOrWhiteSpace(x["StampName"].ToString()));
+            if (stampNameObjects.Any())
+                return stampNameObjects.First()["StampName"].ToString();
+
+            throw new Exception($"Admin Sites response did not contain stamp name for site {siteName}");
```

Comment:
In what conditions do we run into this?

Reply:
Lately we've been seeing an issue in which the call to `/adminsites` succeeds, but this method was unable to retrieve the stamp name from the response body.